packages/edit-navigation/src/components/menu-editor/use-stub-post.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * WordPress dependencies
+ */
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import createNavigationBlock from './create-navigation-block';
+
+export default function useStubPost( query ) {
+	const [ postId, setPostId ] = useState( false );
+	const menuItems = useFetchMenuItems( query );
+	const { receiveEntityRecords } = useDispatch( 'core' );
+	const { setMenuItemsToClientIdMapping } = useDispatch(
+		'core/edit-navigation'
+	);
+	useEffect( () => {
+		setPostId( null );
+		if ( menuItems === null ) {
+			return;
+		}
+		const [ navigationBlock, menuItemIdToClientId ] = createNavigationBlock(
+			menuItems
+		);
+		setMenuItemsToClientIdMapping( query, menuItemIdToClientId );
+
+		const post = createStubPost( query.menus, navigationBlock );
+		const entityStored = receiveEntityRecords(
+			'root',
+			'postType',
+			post,
+			{ id: post.id },
+			false
+		);
+		entityStored.then( () => {
+			setPostId( post.id );
+		} );
```

Comment:
I am pretty sure this isn't the correct way to store this stub post. Should this even be a post? Maybe a dedicated entity kind would work better?